### PR TITLE
Tests for Store mappings

### PR DESCRIPTION
**Pull request by nu1silva**
Tests for Store mappings

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ *   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java]: License year should be 2017

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/APIMappingUtilTestCase.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
+/*
+ *
+ *   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/APIMappingUtilTestCase.java]: License year should be 2017

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/ApplicationKeyMappingUtilTestCase.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ *   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/ApplicationKeyMappingUtilTestCase.java]: License year should be 2017

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/ApplicationKeyMappingUtilTestCase.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
+/*
+ *
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/ApplicationKeyMappingUtilTestCase.java]: Unwanted blank line

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/ApplicationMappingUtilTestCase.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
+/*
+ *
+ *   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/ApplicationMappingUtilTestCase.java]: License year should be 2017

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/CommentMappingUtilTestCase.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
+/*
+ *
+ *   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/CommentMappingUtilTestCase.java]: License year should be 2017

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java (hunk 1 above)
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java]: fix year

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+import org.wso2.carbon.apimgt.core.models.BusinessInformation;
+import org.wso2.carbon.apimgt.core.models.CorsConfiguration;
+import org.wso2.carbon.apimgt.core.models.policy.Policy;
+import org.wso2.carbon.apimgt.core.models.policy.SubscriptionPolicy;
+import org.wso2.carbon.apimgt.rest.api.publisher.dto.APIDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.dto.API_endpointDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.dto.EndPointDTO;
+import org.wso2.msf4j.Request;
+
+import java.util.*;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java]: remove wild card imports

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        api.setVersion(apiVersion);
+        api.setProvider("provider");
+        api.setDescription("sample descripiton");
+        api.setLifeCycleStatus("PUBLISHED");
+        api.setEndpoint(list1);
+        api.setWsdlUri("http://www.webservicex.net/globalweather.asmx?op=GetWeather?wsdl");
+        api.setResponseCaching("true");
+        api.isDefaultVersion(true);
+        api.setCacheTimeout(120);
+//        api.transport(transport);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java]: remove uncessary commented cdes

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java (hunk 7 above)
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/test/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/TestUtil.java]: Remove the * import

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/APIMappingUtilTestCase.java (hunk 2 above)
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/APIMappingUtilTestCase.java]: fix year

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/APIMappingUtilTestCase.java [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+import org.wso2.carbon.apimgt.core.models.Endpoint;
+import org.wso2.carbon.apimgt.core.models.policy.APIPolicy;
+import org.wso2.carbon.apimgt.core.models.policy.Policy;
+import org.wso2.carbon.apimgt.core.models.policy.SubscriptionPolicy;
+import org.wso2.carbon.apimgt.core.util.APIMgtConstants;
+import org.wso2.carbon.apimgt.rest.api.store.dto.APIDTO;
+import org.wso2.carbon.apimgt.rest.api.store.dto.APIListDTO;
+
+import java.time.LocalDateTime;
+import java.util.*;
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/APIMappingUtilTestCase.java]: Remove the * import

Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/APIMappingUtilTestCase.java]: remove wild card imports

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/ApplicationMappingUtilTestCase.java [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+        applicationDTO.setLifeCycleStatus("PUBLISHED");
+
+        ApplicationMappingUtil applicationMappingUtil = new ApplicationMappingUtil();
+
+        Application application = applicationMappingUtil.fromDTOtoApplication(applicationDTO, "user1");
+
+        Assert.assertEquals(application.getId(), applicationID);
+    }
+
+    @Test
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/ApplicationMappingUtilTestCase.java]: empty test?

**Pull request merged**